bin/user_config_converter.py

```diff
@@ -0,0 +1,166 @@
+from user_config_loaders.platform_1.version_7_1.user_config_loader.user_config_loader import UserConfig as UserConfigTypingClass
+
+from p1_runner import trace as logging
+from p1_runner.argument_parser import ArgumentParser
+
+logger = logging.getLogger('point_one.user_config_converter')
+
+
+class ConversionDirection(Enum):
+    TO_JSON = auto()
+    TO_BINARY = auto()
+
+
+def main():
+    if getattr(sys, 'frozen', False):
+        execute_command = os.path.basename(sys.executable)
+    else:
+        execute_command = os.path.basename(sys.executable)
+        if execute_command.startswith('python'):
+            execute_command += ' ' + os.path.basename(__file__)
+
+    parser = ArgumentParser(
+        usage='%s COMMAND [OPTIONS]... IN_FILE OUT_FILE' % execute_command,
+        description='Convert user configuration files between binary and JSON representations.')
+
+    parser.add_argument('-v', '--verbose', action='count', default=0,
+                        help="Print verbose/trace debugging messages. May be specified multiple times to increase "
+                             "verbosity.")
+
+    parser.add_argument(
+        'in_file',
+        type=Path,
+        help="The path to the file to load. Must have extension '*.json' or '*.p1log'.")
+    parser.add_argument(
+        'out_file',
+        type=Path,
+        help="The path to write the converter file to. If this is a directory, the filename will be based on the input file with new extension. For example /in_dir/bar.json -> /out_file/bar.p1log.")
+
+    options = parser.parse_args()
+
+    if options.verbose == 0:
+        logging.basicConfig(level=logging.INFO, format='%(message)s', stream=sys.stdout)
+    else:
+        logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                            stream=sys.stdout)
+        if options.verbose == 1:
+            logging.getLogger('point_one').setLevel(logging.DEBUG)
+        elif options.verbose > 1:
+            logging.getLogger('point_one').setLevel(logging.getTraceLevel(depth=options.verbose - 1))
+
+    # For type hinting.
+    in_file: Path = options.in_file
+    out_file: Path = options.out_file
+
+    if not in_file.exists():
+        logger.error(f'in_file "{in_file}" not found.')
+        exit(1)
+
+    if in_file.suffix.lower() == '.json':
+        direction = ConversionDirection.TO_BINARY
+    elif in_file.suffix.lower() == '.p1log':
+        direction = ConversionDirection.TO_JSON
+    else:
+        logger.error(f'in_file has unknown file extension "{in_file.suffix}". Only "*.json" or "*.p1log" are accepted.')
+        exit(1)
+
+    if out_file.is_dir():
+        if direction == ConversionDirection.TO_BINARY:
+            extension = '.p1log'
+        else:
+            extension = '.json'
+
+        out_file = out_file / (in_file.stem + extension)
+
+    if direction == ConversionDirection.TO_BINARY:
```

Comment:
Suggested comment for readability: `# Parse the input file and determine the platform type and version of the saved UserConfig data.`